pkg/proxy/iptables/proxier.go

```diff
@@ -552,6 +565,42 @@ func servicePortEndpointChainName(s proxy.ServicePortName, protocol string, endp
 	return utiliptables.Chain("KUBE-SEP-" + encoded[:16])
 }
 
+// getRemovedEndpointIPs returns the endpoint IPs that are missing in the new endpoints
+func getRemovedEndpoints(curEndpoints, newEndpoints []string) []string {
+	var result []string
+	for _, ep := range curEndpoints {
+		if !slice.StringInSlice(ep, newEndpoints) {
+			result = append(result, ep)
+		}
+	}
+	return result
+}
+
+// refreshConnection deletes all service connections specified in the map to avoid hitting dead endpoints
+func (proxier *Proxier) refreshConnection(connectionMap map[string]proxy.ServicePortName) {
+	for endpoint, servicePortName := range connectionMap {
+		svcInfo := proxier.serviceMap[servicePortName]
+		if svcInfo.protocol == api.ProtocolUDP {
```

Comment:
The other thing is that just because an endpoint is removed doesn't imply that the pod is dead.
The tcp connection could still be active.
So we should probably only kill udp.